snapcraft/internal/parser.py

```diff
@@ -134,7 +134,24 @@ def _update_source(part, origin):
     return part
 
 
-def _process_entry_parts(entry_parts, parts, origin, maintainer, description):
+def _replace_variables(item, name, version):
```

Comment:
this looks a lot like `project_loader`'s `replace_attr`. Maybe generalize it a bit to define a list of replacement tuples such that you can pass something like this

```python
[
    ('$SNAPCRAFT_PROJECT_NAME', name),
    ('$SNAPCRAFT_PROJECT_VERSION', version),
]
```

and then in the function for the `str` instances:
```python
for replacement in replacements:
    attr.replace(replacement[0], replacement[1])
```